### ISSUE TITLE
Boolean rowsList or colsList return incorrect Matrix

### ISSUE THREAD

**Issue by mvnnn** [sympy/sympy#10658]
```
>>> m = Matrix(3, 3, range(9))
>>> m.extract([0, 1, 2], [False, True, False])
⎡ 0 1 0 ⎤
| 3 4 3 |
⎣ 6 7 6 ⎦
```
it's return wrong output.when we takes ```boolean``` value for ```rowsList``` or ```colsList```.

**Comment by aravindkanna** [comment]
Now it is returning an error stating

**IndexError: Invalid index a[True]**

so, is the error fixed? Or I would like to continue with this. Can anyone please guide me?

**Comment by mvnnn** [comment]
@aravindkanna that's issue is fixed in PR#10659.
